messages.md

```diff
@@ -0,0 +1,382 @@
+interconnected.
+
+As each wallet will verify and store signatures locally, the server isn't trusted and can be
+managed by the organisation deploying Revault.
+
+All transactions are signed paying a fixed 253perkw feerate.
+
+
+### Rough flow
+
+```
+  WALLET                      SYNC_SERVER
+    ||   -A-- sig   -------->    ||   // A: Here is a sig for this id !
+    ||   -C-- sig   -------->    ||
+    ||   -B-- sig   -------->    ||
+    ||
+    ||   -C-- get_sigs  ---->    ||   // C: I gave my sig but now am waiting for everyone to complete..
+            ...(polling)
+    ||   -A-- get_sigs  ---->    ||
+            ...(polling)
+    ||   -B-- get_sigs  ---->    ||
+            ...(polling)
+    ||   -A-- err_sig   ---->    ||   // A: Huh lol. This sig isn't valid.
+    ||   <-- get_sigs error--    ||   // Server: Someone's is unhappy with the sig so I erased all of them, try again.
+    ||
+    ||   -A-- sig  --------->    ||
+    ||   -B-- sig  --------->    ||
+    ||   -C-- sig  --------->    ||
+    ||
+    ||   -A-- get_sigs  ---->    ||
+    ||   -B-- get_sigs  ---->    ||
+    ||   -C-- get_sigs  ---->    ||
+            ...(polling)
+            ...(polling)              // Eventually they all retrieve the sigs.
+```
+
+### Messages
+
+#### `sig`
+
+Sent by a wallet at any point in time to share the signature for a transaction with
+everyone.
+
+The wallet can safely post its signature for both the `cancel` and `emergency` of each
+`vault` utxo without waiting for others. However, it must wait for everyone to have signed
+the `cancel` and `emergency` transactions and its watchtower to have verified and stored
+the signature before sharing its signature for the unvault transaction.
+
+Revocation transaction (`cancel` and `emergency`s) are signed with the `ALL|ANYONECANPAY`
+flag.
+
+```json
+{
+    "method": "sig",
+    "params": {
+        "signature": "Bitcoin ECDSA signature as hex",
+        "id": "tx uid"
+    }
+}
+```
+
+The `tx uid` is `sha256(txid)`. It's used when polling.
+
+No explicit ACK from the server as the wallet can just `get_sigs` for its own signature.
+
+
+#### `get_sigs`
+
+Sent by a wallet to retrieve all signatures for a specific transaction.
+
+```json
+{
+    "method": "get_sigs",
+    "params": {
+        "id": "tx uuid"
+    }
+}
+```
+
+The server answers with a (possibly incomplete) list of the signatures for this
+transaction.
+
+```json
+{
+    "result": {
+        "signatures": ["sigA", null, "sigC"]
+    }
+}
+```
+
+If a wallet notices its transaction to be absent, it must send it again. It can either
+mean the server didn't store it (no explicit ACK) or someone was unhappy with it (no
+explicit error from the server).
+
+
+#### `err_sig`
+
+Sent by a wallet to express its dreadful unhapiness with one of the returned signatures.
+It results in the server erasing all the signatures for this transaction and make other
+wallets send a new signature.
```

Comment:
what if the key of a wallet is compromised and the attacker spam the sync server with err_sig message ?
In supplement of the sig check, maybe the sync server can distribute authentication token to clients which access can be revoked the time all the stakeholders set a new vault and move fund to it.

Reply:
Regarding auth tokens, all messages are exchanged over an encrypted and authenticated (bolt 8) communication channel.

Regarding blocking of operations, we fundamentally cannot protect it. We could implement a small DOS protection obv, but to block signing operations a wallet could for example just never send the `sig` message.